index.html

```diff
@@ -1,16 +1,77 @@
 <!DOCTYPE html>
 <html lang="en">
-
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
     <title>Responsive Cake webpage</title>
+    <link rel="stylesheet" href="/style.css" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
+    />
     <!-- Add a link to your css file here -->
-</head>
+  </head>
+
+  <body>
+    <main class="index-main">
+      <section class="index-hero">
+        <div class="motto">
+          <img alt="cake logo" src="img/cupcakelogo.jpg" />
+          <p>The best cakes in town delivered to your door</p>
 
-<body>
-    <!-- Add your markup here -->
-</body>
+          <div class="topnav" id="myTopnav">
+            <a href="#home">HOME</a>
+            <a href="#news">CAKES</a>
+            <a href="#contact">ORDERIN</a>
+            <a href="#about">LESSONS</a>
+            <a href="#about">ABOUT</a>
+            <a href="javascript:void(0);" class="icon" onclick="myFunction()">
+              <i class="fa fa-bars"></i>
+            </a>
+          </div>
+        </div>
+      </section>
 
-</html>
\ No newline at end of file
+      <section class="index-description">
+        <h1>Welcome</h1>
+        <p>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+          aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet,
+          consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
+          labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud
+          exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+        </p>
+      </section>
+      <section class="index-description-picture"></section>
+      <section class="index-products">
+        <img alt="cake picture" src="/img/cake1.jpeg" />
+        <img alt="cake picture" src="/img/cake2.jpeg" />
+        <img alt="cake picture" src="/img/cake3.jpeg" />
+        <img alt="cake picture" src="/img/cake4.jpeg" />
+      </section>
+      <footer>
+        <div class="socialMedia">
+          <p>Tweeter</p>
+          <p>Facebook</p>
+          <p>Instagram</p>
+        </div>
+        <p>Copyright © Cake4all Ltd</p>
+      </footer>
+    </main>
+    <script>
+      function myFunction() {
```

Comment:
I am impressed that you're adding JavaScript here to render the nav elements on mobile, and your solution of changing CSS classes works well!

We can toggle CSS classes with JS, so you can reduce the conditional logic and make the code slightly clearer:
```
function toggleNavBar() {
  const element = document.getElementById("myTopnav");
  element.classList.toggle("responsive", "top-nav");
} 
```

myFunction is quite possibly the least descriptive name you could have gone for here! I'd change this to something that actually gives a description of what the function does.

You also need to be careful that you're making images responsive when you manipulate the DOM like this, as I can see the below screenshot where the logo becomes warped.

<img width="370" alt="Screenshot 2022-11-11 at 09 49 34" src="https://user-images.githubusercontent.com/61597016/201314087-d7e7016f-d426-4cd7-b5bf-c14693e498a1.png">

Finally, it might interest you to look into CSS transition animations. It's a really cool feature of CSS that lets you gradually ease DOM changes in without much effort at all, rather than just instantly snapping the DOM into a new state: https://www.w3schools.com/css/css3_transitions.asp

I think this is a really cool thing to have added though, and great work on getting ahead on the JS.